src/server/health_check.go

```diff
@@ -0,0 +1,37 @@
+package server
+
+import "net/http"
+
+// HealthCheck defines a readiness check for a Webhook struct based on the presence of its TLS certificate and key.
+// It requires the whole webhook as parameter to be able to RLock on the certificate for the presence confirmation.
+type HealthCheck struct {
+	webhookServer *Webhook
+}
+
+// ServeHTTP checks if the Webhook server has a TLS certificate/key pair.
+func (h *HealthCheck) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.webhookServer.RLockCert()
+	defer h.webhookServer.RUnlockCert()
+
+	if h.webhookServer.Cert == nil {
+		var response = "Certificate not present."
+		w.WriteHeader(503)
+		if _, err := w.Write([]byte(response)); err != nil {
+			h.webhookServer.Logger.Errorw("can't write response", "err", err, "response", response)
+		}
+		return
+	}
+
+	var okResponse = "OK"
+	w.WriteHeader(200)
```

Comment:
You don't need to write the 200 header here as it will be done by the `w.Write`. See https://golang.org/src/net/http/server.go